nequip/model/_eng.py

```diff
@@ -13,14 +14,22 @@
     SphericalHarmonicEdgeAttrs,
 )
 
+from . import builder_utils
 
-def EnergyModel(config) -> SequentialGraphNetwork:
+
+def EnergyModel(
+    config, initialize: bool, dataset: Optional[AtomicDataset] = None
+) -> SequentialGraphNetwork:
     """Base default energy model archetecture.
 
     For minimal and full configuration option listings, see ``minimal.yaml`` and ``example.yaml``.
     """
     logging.debug("Start building the network model")
 
+    builder_utils.add_avg_num_neighbors(
+        config=config, initialize=initialize, dataset=dataset
+    )
+
```

Comment:
does this config revision happen after `trainer.__init__` ?

Reply:
yes, is in a model builder

Reply:
but it gets written out to the saved config.yaml:
```
avg_num_neighbors: 10.495238304138184
```